stock_picking_progress/models/stock_move_line.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import api, fields, models
+from odoo.tools.float_utils import float_is_zero
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    progress = fields.Float(compute="_compute_progress", store=True)
+
+    @api.depends(
+        "product_uom_qty", "product_uom_id", "product_uom_id.rounding", "qty_done"
+    )
+    def _compute_progress(self):
+        for record in self:
+            rounding = record.product_uom_id.rounding
+            if float_is_zero(record.product_uom_qty, precision_rounding=rounding):
+                record.progress = 0.0
+            else:
+                record.progress = (record.qty_done / record.product_uom_qty) * 100
```

Comment:
In order to be used by the widget `progressbar`/`percentage`/`percentpie` you can't do that. Keep this operation when rendering the value for the user.
```suggestion
                record.progress = (record.qty_done / record.product_uom_qty)
```
Maybe rename this field `progress_ratio` then?

Reply:
progressbar accepts values as percentages.

Reply:
I was wrong about the `* 100`, by default Odoo expects a %, not a ratio, for its different widgets.